index.php

```diff
@@ -250,138 +250,215 @@
+        // Get a string for this sort link.
+        $columnheader = get_string("table:sort_header_$sortcolumn", 'report_coursecompletion');
+        // Update parameters for sort and direction for this column in the final url.
+        $baseurl->param('sort', $sortcolumn);
+        $baseurl->param('dir', $columndir);
+        if ($sortcolumn === "completionstatus") {
+            $final[] = "<a href=javascript:void(0)>$columnheader</a>";
         } else {
-            $final[] = "<a href=$base_url#table>$column_header</a>$column_icon";
+            $final[] = "<a href=$baseurl#table>$columnheader</a>$columnicon";
         }
     }
-    //If one column has multiple sorts, combine them into one entry for that column.
+    // If one column has multiple sorts, combine them into one entry for that column.
     $hcolumns[$column] = implode('/', $final);
 }
 
 $table = new html_table();
 $table->head = $hcolumns;
 $table->attributes["class"] = "admintable generaltable";
 $table->data = [];
-foreach($records as $record) {
+foreach ($records as $record) {
     $final = new stdClass();
-    if (IS_ADMIN) {
+    if (REPORT_COURSECOMPLETION_IS_ADMIN) {
         $finalname = fullname($record);
-        $final->fullname = add_link('/user/view.php', array('id' =>$record->userid), $finalname);
+        $final->fullname = add_link('/user/view.php', array('id' => $record->userid), $finalname);
         $final->email = $record->email;
     }
-    $final->course = add_link('/course/view.php', array('id' =>$record->course), $record->fullname);
+    $final->course = add_link('/course/view.php', array('id' => $record->course), $record->fullname);
     $final->timestarted = time_format($record->timestarted);
     $final->timecompleted = time_format($record->timecompleted);
     $final->completionstatus = $record->timecompleted ? 'Yes' : 'No';
     $table->data[] = $final;
 }
 
 echo html_writer::table($table);
-$buttonurl = new moodle_url("index.php", array("sort" => $sort, "dir" => $dir, "perpage" => $perpage, "page" => $page, "export"=>1));
+$buttonurl = new moodle_url("index.php", array(
+    "sort" => $sort,
+    "dir" => $dir,
+    "perpage" => $perpage,
+    "page" => $page,
+    "export" => 1
+    )
+);
 $buttonstring = get_string('exportbutton', 'report_coursecompletion');
 echo $OUTPUT->single_button($buttonurl, $buttonstring);
 echo $OUTPUT->footer();
 
-function add_condition_connectors(&$data, &$where, &$params, $force_and = false) {
-    if(!empty($params)) {
-        if($force_and || (isset($data->operator) && $data->operator == 0)) {
+/**
+ * Appends AND / OR operators to data fields that are assigned to $where param.
+ * @param object $data Form data stored in session.
+ * @param string $where Where clause.
+ * @param array $params Typically stores user ID and cohort ID.
+ * @param string $forceand A nonempty value would convert an OR operator
+ * to AND.
+ */
+function add_condition_connectors(
```

Comment:
functions should not be hidden at the bottom of inline php files - they should be in a locallib file or a class within the classes folder and should also follow the function naming rules. "add_condition_connectors" is not a valid function name within a plugin.